crates/sputnik/src/session.rs

```diff
@@ -133,9 +140,14 @@ impl Session {
 
     /// sends anonymous usage data to the endpoint defined in ReportingInfo.
     pub fn report(&self) -> Result<(), SputnikError> {
-        if self.reporting_info.is_telemetry_enabled && !cfg!(debug_assertions) {
-            // set timeout to 400 ms to prevent blocking for too long on reporting
-            let timeout = Duration::from_millis(4000);
+        // Disable telemetry by default when not a production release.
+        // TODO: consider whether we want to disable non-production telemetry or at least document
+        // the reasoning for not using it
+        if !cfg!(debug_assertions) && !cfg!(test) {
```

Comment:
Since this isn't inherently related to sending arch/os metrics, can we scoop this out and create an issue on how we might make this more malleable for debug/test paths? It seems like something where we would want to be able to inject a mock, or provide a non-production path for testing.

Reply:
up to you; I added the `!cfg!(test)` to make sure I could write tests for the report() fn, but the `!cfg!(debug_assertions)` screams for something injectable

I definitely wouldn't want to remove the debug_assertions check or the tests below (mostly to test the timeout functionality and have a good signal that the expected headers get sent)

Reply:
Looking back at this, I think we should leave this as you have it. I (unsurprisingly) didn't see the original code that contained `!cfg!(debug_assertions)` and assumed that it was something new.

However, (this is a nit) to clean up the return paths (since it's doing an early return and then another conditional), we might want to consider doing something like this:
```
let should_send_report = !cfg!(debug_assertions) && !cfg!(test) && self.reporting_info.is_telemetry_enabled;
if should_send_report {
...
}
Ok(())
```

Reply:
sgtm; will make that change!

and yeah, I really want to understand _why_ we're skipping prod releases for reporting; legal thing? devex thing? thing I'm misunderstanding? mystery upon mystery

Reply:
yeah, that's way cleaner, thank you

Reply:
dang it, couldn't figure out a good way to get the conjucts to represent what we want; tests will fail because the `!cfg!(test)` will always be true during testing, meaking `should_send_report` always be false (meaning testing the timeout and success paths can't happen)

so, going to keep as-is 😞 way more readable the other way, though